modules/islandora_videojs/islandora_videojs.module

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/**
+ * @file
+ * Contains islandora_videojs.module.
+ *
+ * This file is part of the Islandora Project.
+ *
+ * (c) Islandora Foundation
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Drupal\Core\Routing\RouteMatchInterface;
+
+/**
+ * Implements hook_help().
+ */
+function islandora_videojs_help($route_name, RouteMatchInterface $route_match) {
+  switch ($route_name) {
+    // Main module help for the islandora module.
+    case 'help.page.islandora_videojs':
+      $output = '';
+      $output .= '<h3>' . t('About') . '</h3>';
+      $output .= '<p>' . t('Islandora Videojs overrides.') . '</p>';
+      $output .= '<p>' . t('Customizes videojs.') . '</p>';
+      return $output;
+
+    default:
+  }
+}
+
+function islandora_videojs_theme_registry_alter(&$theme_registry) {
+  $module_path = drupal_get_path('module', 'islandora_videojs');
+
+  // Use the templates in my module's template folder.
+  $theme_registry['videojs'] = $theme_registry['videojs'];
+  $theme_registry['videojs']['path'] = $module_path . '/templates';
+  $theme_registry['videojs']['template'] = 'videojs';
+}
+
+
+/**
+* videojs theme override
+*/
+function islandora_videojs_theme($existing, $type, $theme, $path) {
+  $transcript_urls = get_transcript_urls();
+  return array(
+    'videojs' => array(
+      'variables' => array('items' => NULL, 'player_attributes' => NULL, 'mimes' => NULL, 'transcript_urls' => $transcript_urls),
+      'base hook' => 'videojs'
+    ),
+  );
+}
+
+/**
+* returns the urls of the transcript of a repository item
+*/
+function get_transcript_urls() {
+  // Get the nid
+  $node = \Drupal::routeMatch()->getParameter('node');
+  $nid = NULL;
+  if ($node instanceof \Drupal\node\NodeInterface) {
+    $nid = $node->id();
+  }
+
+  $transcript_urls = array();
+
+  if ($nid != NULL) {
```

Comment:
You can avoid the Guzzle work by using IslandoraUtils. After you [lookup the Term ID](https://github.com/Islandora/islandora/blob/8.x-1.x/src/IslandoraUtils.php#L223-L245), e.g. `$transcript_term = getTermForUri('http://pcdm.org/use#Transcript');`,  take your `$node` from the routeMatch and use [getMediaWithTerm](https://github.com/Islandora/islandora/blob/8.x-1.x/src/IslandoraUtils.php#L164-L186) to find your Transcript Media, e.g. `$media = getMediaWithTerm($node, $transcript);`.

Reply:
Between `IslandoraUtils` and the `MediaSourceService` you can replace huge chunks of this.